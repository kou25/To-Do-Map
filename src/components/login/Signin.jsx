import React, { Component } from 'react'
import { Link } from 'react-router-dom';


 class Signin extends Component {
    state={
            email:'',
            password:''
        }
        handleChange=(e)=>{
            this.setState({
                [e.target.id]: e.target.value
            })
        }
        handleSubmit=(e)=>{
            e.preventDefault()
            console.log(this.state)
            this.props.history.push('/');
    }

    render() {
        return (
            <div className="container">
                <div data-testid="heading" className="d-sm-flex align-items-center justify-content-between mb-4">
                    <h1 style={{color:"#8B4513",  fontFamily:"cursive", margin:"30px 30%"}}>Sign In<i className="fa fa-sign-in"  style={{marginLeft:"20px", color:"#8B4513"}}></i></h1>
                </div>
                <form onSubmit={this.handleSubmit} style={{width:"60%",fontFamily:"cursive", marginLeft:"10%"}}>
                <div className="form-group">
                    <label htmlFor="exampleInputEmail1">Email address</label>
                    <input type="email" 
                    className="form-control" 
                    placeholder="Enter email"
                    id="email"
                    onChange={this.handleChange}
                    required/>
                </div>
                <div className="form-group">
                    <label htmlFor="exampleInputPassword1">Password</label>
                    <input type="password" 
                    className="form-control" 
                     placeholder="Password"
                     id="password"
                     onChange={this.handleChange}
                     required/>
                </div>
                
                <button data-testid="button" type="submit" className="btn btn-warning mr-2 shadow"  
                >Sign In</button>
                <Link to ={'/signup'}>
                <button data-testid="button1" type="submit" className="btn btn-primary mr-2 shadow"
                >Sign Up</button></Link>
                <Link to ={'/forget'}>
                <button data-testid="button2" type="submit" className="btn btn-danger shadow"
                >Forget Password</button></Link>
                </form>
            </div>
        )
    }
}


export default Signin;