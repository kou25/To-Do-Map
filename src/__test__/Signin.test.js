import React from 'react';
import ReactDom from 'react-dom'
import Signin from '../components/login/Signin'
import { BrowserRouter } from 'react-router-dom';
import {render, cleanup} from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer'

afterEach(cleanup);
it("render without crashing",()=>{
  const div= document.createElement("div");
  ReactDom.render(
  <BrowserRouter>
  <Signin/>
  </BrowserRouter>,div);
})


  

  it("renders button correctly", ()=>{
    const{getByTestId}=render(
    <BrowserRouter>
      <Signin/>
      </BrowserRouter>)
    expect(getByTestId('button')).toHaveTextContent('Sign In')
  })

  it("renders button correctly", ()=>{
    const{getByTestId}=render(
    <BrowserRouter>
      <Signin/>
      </BrowserRouter>)
    expect(getByTestId('button1')).toHaveTextContent('Sign Up')
  })

  it("renders button correctly", ()=>{
    const{getByTestId}=render(
    <BrowserRouter>
      <Signin/>
      </BrowserRouter>)
    expect(getByTestId('button2')).toHaveTextContent('Forget')
  })

  it('matches snapshot', ()=>{
    const tree=renderer.create(
    <BrowserRouter>
      <Signin/>
      </BrowserRouter>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  