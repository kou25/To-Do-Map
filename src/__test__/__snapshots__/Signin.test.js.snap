// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="d-sm-flex align-items-center justify-content-between mb-4"
    data-testid="heading"
  >
    <h1
      style={
        Object {
          "color": "#8B4513",
          "fontFamily": "cursive",
          "margin": "30px 30%",
        }
      }
    >
      Sign In
      <i
        className="fa fa-sign-in"
        style={
          Object {
            "color": "#8B4513",
            "marginLeft": "20px",
          }
        }
      />
    </h1>
  </div>
  <form
    onSubmit={[Function]}
    style={
      Object {
        "fontFamily": "cursive",
        "marginLeft": "10%",
        "width": "60%",
      }
    }
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="exampleInputEmail1"
      >
        Email address
      </label>
      <input
        className="form-control"
        id="email"
        onChange={[Function]}
        placeholder="Enter email"
        required={true}
        type="email"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="exampleInputPassword1"
      >
        Password
      </label>
      <input
        className="form-control"
        id="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
      />
    </div>
    <button
      className="btn btn-warning mr-2 shadow"
      data-testid="button"
      type="submit"
    >
      Sign In
    </button>
    <a
      href="/signup"
      onClick={[Function]}
    >
      <button
        className="btn btn-primary mr-2 shadow"
        data-testid="button1"
        type="submit"
      >
        Sign Up
      </button>
    </a>
    <a
      href="/forget"
      onClick={[Function]}
    >
      <button
        className="btn btn-danger shadow"
        data-testid="button2"
        type="submit"
      >
        Forget Password
      </button>
    </a>
  </form>
</div>
`;
